### DISCUSSION TITLE
How would I go about using this with an already existing Database?

### DISCUSSION THREAD

**Discussion by nikolaischunk** [nicoalbanese/kirimase#146] [Q&A] [answered]
I have a database on Planetscale already that has valuable data. 
I am wondering, how I could integrate Kirimase with this existing database?

**Comment by trevorpfiz** [comment]
> I have a database on Planetscale already that has valuable data. I am wondering, how I could integrate Kirimase with this existing database?

@nikolaischunk  you would just set up the project with the Planetscale options and use the env variables from your existing database

**Reply by nikolaischunk** [reply]
> @nikolaischunk you would just set up the project with the Planetscale options and use the env variables from your existing database

I am more referring to the data type definition you can generate with `kirimase generate`.
Do I have to replicate my database schema with the cli options? 
Or how else can I provide Kirimase with the details of how my existing database is structured?

**Reply by trevorpfiz** [reply, marked as answer]
ahh I see. I think you have to replicate the schema with the cli options currently if you want the Models, Controllers, and Views. I had to add one field and went through the whole process again. Partly the limits of code generation without AI I would imagine.

**Reply by nikolaischunk** [reply]
I would love to have something like `kirimase migrate` that uses a provided schema and builds Models, Controllers, and Views on top of it ⚡
